### PR TITLE
hotfix release workflow

### PR DESCRIPTION
**Pull request by theguy951357**
## Checklist before requesting a review
- [x] My changes affect only 10 files or less.
- [ ] I have performed a self-review of my code and tested it.
- [x] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [ ] If it is a bugfix, my code is <= 200 lines.
- [x] My functions are <= 80 lines.
- [x] I have had my code reviewed by a peer.
- [ ] My unit tests pass locally.
- [x] Netclient & Netmaker are awesome.

### CODE REVIEW

**Review by 0xdcarns — APPROVED** [review]

**Review by mattkasun — APPROVED** [review]

File: .github/workflows/hotfix-release.yml [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        if: ${{ github.event.inputs.prerelease == 'true'}}
+        uses: goreleaser/goreleaser-action@v4
+        with:
+          args: release --clean --release-notes release.md -f .goreleaser.prerelease.yaml
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN  }}
+        
+  freebsd-headless:
+    runs-on: ubuntu-latest
+    needs: release-branch
```
Comment (Aceix) [inline .github/workflows/hotfix-release.yml:70]: same

File: .github/workflows/hotfix-release.yml [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          file: /tmp/netclient
+          tag: ${{ github.event.inputs.version }}
+          overwrite: true
+          asset_name: netclient-freebsd-amd64
+
+
+  docker:
+    needs: release-branch
```
Comment (Aceix) [inline .github/workflows/hotfix-release.yml:112]: same

File: .github/workflows/hotfix-release.yml [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          platforms: linux/amd64, linux/arm64, linux/arm/v7
+          push: true
+          tags: gravitl/netclient:${{ github.event.inputs.version }}
+  
+  packages:
+    if: ${{ github.event.inputs.prerelease == 'false' }}
+    needs: release-branch
```
Comment (Aceix) [inline .github/workflows/hotfix-release.yml:138]: same

File: .github/workflows/hotfix-release.yml [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
+      - name: setup git
+        run: |
+          git config user.name "Github Actions"
+          git config user.email "info@netmaker.io"
+      - name: create release branch
+        run: |
+          git switch -c release_${{ github.event.inputs.version }}
```
Comment (Aceix) [inline .github/workflows/hotfix-release.yml:32]: `-c` flag not required since the release branch will already be present in a hotfix scenario

**Review by Aceix — CHANGES_REQUESTED** [review]

File: .github/workflows/hotfix-release.yml [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+      - name: update tag
+        run: |
+          git tag -f ${{ github.event.inputs.version }}
+          git push
+  
+  linux-headless:
+    runs-on: ubuntu-latest
+    needs: release-branch
```
[the remark this answers is not in the document]
Comment (Aceix) [reply .github/workflows/hotfix-release.yml:40]: change to `update-tag`?

**Review by Aceix — APPROVED** [review]

**Pull request merged**